File: php/features.php
<?php
declare(strict_types=1);

// Opensensemap SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class OpensensemapFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new OpensensemapBaseFeature();
            case "test":
                return new OpensensemapTestFeature();
            default:
                return new OpensensemapBaseFeature();
        }
    }
}

File: py/opensensemap_sdk/features.py
# Opensensemap SDK feature factory

from opensensemap_sdk.feature.base_feature import OpensensemapBaseFeature
from opensensemap_sdk.feature.test_feature import OpensensemapTestFeature


def _make_feature(name):
    features = {
        "base": lambda: OpensensemapBaseFeature(),
        "test": lambda: OpensensemapTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
